crates/tarball/src/lib.rs

```diff
@@ -29,25 +41,23 @@ pub async fn download_and_extract(
     let extract_location = cache_directory.join(name).join(version);
 
     let mut stream =
-        reqwest::get(url).await.or(Err(TarballError::Network(url.to_owned())))?.bytes_stream();
+        reqwest::get(url).await.map_err(|e| TarballError::Network(e))?.bytes_stream();
 
     let mut file = File::create(&tarball_location)
-        .map_err(|_| TarballError::FileSystem("failed to create file".to_owned()))
-        .unwrap();
+        .map_err(|e| TarballError::Io(e))?;
 
     while let Some(item) = stream.next().await {
-        let chunk = item.map_err(|_| TarballError::Network("error while downloading file".to_owned()))?;
-        file.write_all(&chunk).map_err(|_| TarballError::FileSystem("error while writing to file".to_owned()))?;
+        let chunk = item.map_err(|e| TarballError::Network(e))?;
+        file.write_all(&chunk).map_err(|e| TarballError::Io(e))?;
     }
 
-    let tar_gz = File::open(&tarball_location).unwrap();
+    let tar_gz = File::open(&tarball_location)?;
     let tar = GzDecoder::new(tar_gz);
     let mut archive = Archive::new(tar);
-    archive.unpack(&extract_location).unwrap();
+    archive.unpack(&extract_location)?;
 
     std::fs::remove_file(&tarball_location)
-        .map_err(|_| TarballError::FileSystem("removing tarball failed".to_owned()))
-        .unwrap();
+        .map_err(|e| TarballError::Io(e))?;
```

Comment:
Do we need this line? If I remember correctly, couple of commits earlier you added `?` at the end of the fs function, and removed `or`, which magically converted this functions error code to appropriate error?

Example: `std::fs::create_dir_all(&node_modules)?;`

Reply:
It is possible we don't; I was more focused on mechanical transformations than stepping back a bit, so it is quite possible we do not.